### PR TITLE
TS-28 - adding wfh event and hook up endpoint

### PR DESCRIPTION
**Pull request by lee-cheung**
commits: 
Just one commit. adding wfh event and hook up endpoint

screenshot_1540311136
screenshot_1540311145

endpoint: /workingFromHome/findByEmployeeId/{employeeID}
GET request returns: 

`[
  {
    "eventId": 58,
    "startDate": [
      2018,
      10,
      24
    ],
    "endDate": [
      2018,
      10,
      24
    ],
    "eventStatusId": 1,
    "eventStatusDescription": "Awaiting approval",
    "employee": {
      "employeeId": 8,
      "forename": "homer",
      "surname": "simpson",
      "email": "homer@test.com",
      "totalHolidays": 4,
      "startDate": [
        2018,
        10,
        22
      ],
`

WFH is displayed as 'Awaiting Approval' for the mean time. I will add a story on JIRA to update UI so that we are able to distinguish between annual leave and wfh events .

### CODE REVIEW

File: src/components/Booking/EventTypeGroup/index.js [hunk 1]
```diff
@@ -1,23 +1,33 @@
 import React, { Component } from 'react';
 import { View, StyleSheet, Text, TouchableOpacity } from 'react-native';
+import { PropTypes as PT } from 'prop-types';
 import { Icon } from 'react-native-elements';
 import { GREY, WHITE } from '../../../styles/colors';
+import type from '../../../constants/eventTypes';
 
 class EventTypeGroup extends Component {
+  static propTypes = {
```
Comment (d1wilko) [inline src/components/Booking/EventTypeGroup/index.js:9]: not a huge deal, but do we have a standard place to put proptypes? closer to the bottom of the file?

File: src/components/Booking/EventTypeGroup/index.js [hunk 2]
```diff
@@ -1,23 +1,33 @@
...
+
   constructor() {
     super();
     this.state = {
       eventType: [
-        { type: 'Annual leave', icon: 'suitcase', color: '#A7BF35' },
-        { type: 'Working from Home', icon: 'home', color: '#399BB6' },
+        { type: type.ANNUAL_LEAVE, icon: 'suitcase', color: '#A7BF35' },
+        { type: type.WFH, icon: 'home', color: '#399BB6' },
         { type: 'Sick leave', icon: 'bed', color: '#A2798F' },
```
Comment (d1wilko) [inline src/components/Booking/EventTypeGroup/index.js]: worth adding these 2 other types to the constants while you are in the code?

**Review by d1wilko — CHANGES_REQUESTED** [review]
2 small issues - otherwise looks good to me, but I would recommend a second set of eyes, its early haha

File: src/components/Booking/EventTypeGroup/index.js (hunk 1 above)
Comment (lee-cheung) [reply src/components/Booking/EventTypeGroup/index.js:9]: we only put propTypes at bottom of file if it is a stateless class

File: src/components/Booking/EventTypeGroup/index.js (hunk 2 above)
Comment (lee-cheung) [reply src/components/Booking/EventTypeGroup/index.js]: yes, i will make that change

File: src/components/Booking/EventTypeGroup/index.js (hunk 1 above)
Comment (d1wilko) [reply src/components/Booking/EventTypeGroup/index.js:9]: cool, thats fair

File: src/components/Booking/container.js [hunk 3]
```diff
@@ -55,6 +58,7 @@ export default Container => class extends Component {
         endDate: holiday.endDate || chosenDate,
         halfDay: holiday.halfDay,
         duration: holiday.duration,
+        eventType: holiday.eventType === undefined ? eventType.ANNUAL_LEAVE : holiday.eventType,
```
Comment (d1wilko) [inline src/components/Booking/container.js]: i think here you can just do the following `eventType: holiday.eventType || eventType.ANNUAL_LEAVE,`

**Review by d1wilko — CHANGES_REQUESTED** [review]
one small fix and then I'm happy to approve

File: src/components/Booking/container.js (hunk 3 above)
Comment (lee-cheung) [reply src/components/Booking/container.js]: yes, you're right. i have made the change :)

**Review by d1wilko — APPROVED** [review]
looks good to me, try to get another set of eyes on this before merging :)

**Review by mcromie92 — APPROVED** [review]
looks good to me :)

**Pull request merged**